### DISCUSSION TITLE
Primer/CSS is deprecated

### DISCUSSION THREAD

**Discussion by dometto** [gollum/gollum#1999] [General]
It seems like the `primer/css` library is being abandoned by GitHub:

https://primer.style/design/deprecated-components/select-menu/css/

Alternatives are `primer/react` and `primer/view-components`. However, neither seems to be very easy to implement in Gollum. ViewComponent seems to be tied very deeply to Rails, whereas using `react` would, I think, require substantial changes to gollum's architecture.

Short term, I propose that we just keep using `primer/css`, and improve the UI as well as we can within that framework. Longer term, there may be some things we simply cannot achieve using `primer/css` (such as Dialogs?). To reduce reliance on our JS (for e.g. the dialog), we might then have to migrate to a different CSS framework.

@benjaminwil: do you have any ideas about this? Do you agree that using `primer/react` or `primer/view-components` is not feasible? Longer term, would moving to an alternative framework like tailwind be an option?

**Comment by benjaminwil** [comment]
With the caveat being that I have never used view components before, I think that if we were to continue to use Primer, `primer/view_components` would be more straightforward to integrate with the existing Gollum frontend code. How straightforward, though, I'm not sure.

I have experience writing React. While I don't hate it, I don't think it makes sense for Gollum to pivot to using React just for Primer. This pivot would be a lot of work, I think.

That said, migrating to any CSS-first frontend framework would likely be a lot of work 😪

Thanks for opening this discussion and mentioning me. I will think more on this.

**Reply by dometto** [reply]
> That said, migrating to any CSS-first frontend framework would likely be a lot of work 😪

Yes :-( I think my preferences would be:

1. Short term: just stick with `primer/css` and make as many front end improvements as we can with that.
2. Longer term: find a way to include primer ViewComponents.

I've tried just plugging in the `primer-view_components` gem into gollum, but it seems heavily and unreasonably reliant on Rails. As far as I can see, ViewComponents are really only a set of Ruby methods that generate HTML snippets. Maybe someone can or will put together a Sinatra adaptation.
